src/waveform/renderers/glwaveformrenderbackground.cpp

```diff
@@ -0,0 +1,25 @@
+#include "waveform/renderers/glwaveformrenderbackground.h"
+
+GLWaveformRenderBackground::GLWaveformRenderBackground(
+        WaveformWidgetRenderer* waveformWidgetRenderer)
+        : WaveformRenderBackground(waveformWidgetRenderer) {
+}
+
+GLWaveformRenderBackground::~GLWaveformRenderBackground() {
+}
+
+#if !defined(QT_NO_OPENGL) && !defined(QT_OPENGL_ES_2)
+void GLWaveformRenderBackground::draw(QPainter* painter, QPaintEvent* /*event*/) {
+    maybeInitializeGL();
+
+    painter->beginNativePainting();
+    glClearColor(static_cast<float>(m_backgroundColor.redF()),
+            static_cast<float>(m_backgroundColor.greenF()),
+            static_cast<float>(m_backgroundColor.blueF()),
+            1.f);
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+    painter->endNativePainting();
+
+    drawImage(painter);
```

Comment:
why the drawImage?

Reply:
Yeah, I'd prefer if the control flow was a little more visible here. I think it would be best if split into `drawImage` and `drawFlatColor` and a `shouldDrawImage`. Then plain `draw` would just be `if(shouldDrawImage()) {drawImage();} else {drawFlatColor()}; IMO thats pretty legible while still having clearly separated concerns.

Reply:
only that in the case of gl the glclear should always be done. but i'll add the conditional.

Reply:
done